internal/controller/postgrescluster/controller.go

```diff
@@ -144,6 +145,10 @@ func (r *Reconciler) Reconcile(
 		return result, nil
 	}
 
+	if cluster.Spec.OpenShift == nil {
+		cluster.Spec.OpenShift = &r.IsOpenshift
```

Comment:
🔧 This should move to ~L109 so that it doesn't get sent in patches that update the PostgresCluster status.

Failing test shows `\"f:spec\":{\"f:openshift\":{}}` in the managed fields.

Reply:
Wow that makes sense, I will move it. I fought with this for a while (just now seeing your comment this morning) before I figured out the following
```
	if cluster.Spec.OpenShift == nil && r.IsOpenshift {
		cluster.Spec.OpenShift = &r.IsOpenshift
	}
```
I'm thinking with this the openshift will only automatically be set if its true.

Reply:
That would avoid the failure in Kubernetes and trigger the failure in OpenShift? I'd prefer that we treat both environments the same.